android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt

```diff
@@ -309,10 +309,29 @@ class ChargebeeFlutterSdkPlugin : FlutterPlugin, MethodCallHandler, ActivityAwar
 fun CBProduct.toMap(): Map<String, Any> {
     return mapOf(
         "productId" to productId,
-        "productPrice" to productPrice,
+        "productPrice" to skuDetails.priceAmountMicros / 1_000_000.0,
+        "productPriceString" to productPrice,
         "productTitle" to productTitle,
-        "currencyCode" to skuDetails.priceCurrencyCode
+        "currencyCode" to skuDetails.priceCurrencyCode,
+        "subscriptionPeriod" to subscriptionPeriod()
     )
 }
 
+ fun CBProduct.subscriptionPeriod(): Map<String, Any> {
+    val subscriptionPeriod = skuDetails.subscriptionPeriod
+    val numberOfUnits = subscriptionPeriod[subscriptionPeriod.length / 2].digitToInt()
```

Comment:
Can the value be "P10D"? If so can we make sure if handle for it.

Reply:
I believe the subscription with 10/11months wouldn't be possible in play store billing period.

Reply:
From the docs [here](https://support.google.com/googleplay/android-developer/answer/140504?hl=en#zippy=%2Ccreate-and-activate-a-base-plan), I see that the possible values are: 
```
Billing period: Select the duration of the subscription entitlement. The available billing periods are:

    1 week
    4 weeks
    1 month
    3 months
    6 months
    1 year
```
Still would be good practice to parse this so that if a new period is introduced like P15D, it doesn't break this.